### ISSUE TITLE
Is there oneline playground?

### ISSUE THREAD

**Issue by summer-wu** [snowcoders/sortier#1463]
Is there online playground? Like [prettier playground](https://prettier.io/playground/)

**Comment by k2snowman69** [comment]
TL;DR: It's in the works

Long version:
I started on one a while ago but kept running into build issues due to how the sortier library was built. That plus the new ESM support across the page it was more efficient for me to put it on hold until I rebuild this project in ESM to get better treeshaking when setting up the website. I finished up the ESM port (sortier@1.0.0-beta.7) just a few days ago and I really want to pick this back up soon.

I'm going to leave this ticket open and use it as my tracker for getting it done since it was more of a trial thing on my local machine at the time and I never created an official ticket for it.

No ETA yet, I need to do some investigation but with the Christmas holidays coming up I'm hopeful I can get some time to dig into it again.

**Comment by D1no** [comment]
Wouldn't a [codesandbox](codesandbox.io) be the best / easiest way? That's pretty quick to set-up and also a great way to validate real-world functionality.

PS: @k2snowman69 I really really love your library! Really helps me getting through spikes on personal projects ❤️

**Comment by k2snowman69** [comment]
@D1no Thanks for the kind words! And 100% agree that a codesandbox would work to solve this problem. My intention with the playground was to have people reproduce bugs directly in the sandbox and when issues do arise, be able to copy paste the url in the github issue so it's easy to reproduce and validate it's been fixed. 

Also I won't lie and it's also a self guided challenge  to see if I can replicate a similar playground to prettier and typescript's but with less maintenance. That is delaying things so if the priority of this ticket gets bumped higher I'm happy to change course.

For a status update, last I left off was last weekend where I was fighting typescript-eslint looking for node libraries that I'll need to polyfill in (similar to what prettier needs to do here https://github.com/prettier/prettier/blob/main/website/webpack.config.js).

If people are interested in contributing, I'd be happy to push up the branch and create a checklist of TODO items left. My only request is be realistic if you actually plan to help or not... it's a decent amount of effort to plan.

**Comment by k2snowman69** [comment]
So a very initial draft version of the playground can be found here:
https://snowcoders.github.io/sortier/playground/index.html

There is still quite a bit to do before this will be 100% up and running such as
 - Use Monaco or CodeMirror to get a better editing experience with code syntax highlighting
 - Improve the css of the playground in general
 - Implement the options bar on the left
 - Use url state to store options/input so it can be used as a replication environment

I mostly loaded it with what I have so far to do some initial perf testing but since people in here are looking for a playground I figured I could post it to give first looks.
